lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -197,8 +197,16 @@ private static function get_inner_blocks_from_navigation_post( $attributes ) {
 			// it encounters whitespace. This code strips it.
 			$compacted_blocks = gutenberg_block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
 
+			// Run Block Hooks algorithm to inject hooked blocks.
+			$markup = gutenberg_hook_first_last_children_into_core_navigation( $compacted_blocks, $navigation_post->ID );
+			$blocks = parse_blocks( $markup );
+
 			// TODO - this uses the full navigation block attributes for the
 			// context which could be refined.
+			if ( isset( $blocks[0]['innerBlocks'] ) ) {
+				return new WP_Block_List( $blocks[0]['innerBlocks'], $attributes );
```

Comment:
For me, this conditional definitely needs a comment, as the prev line refers to the line after.

Reply:
Hmm, it seems like both this `return` and the one below use `$attributes` though, so the TODO refers to both of them?

One thing we could do to avoid this is something like

```php
$blocks = isset( $blocks[0]['innerBlocks'] ) ? isset( $blocks[0]['innerBlocks'] ) : $compacted_blocks;
// TODO - this uses the full navigation block attributes for the
// context which could be refined.
return new WP_Block_List( $blocks, $attributes );
```

Reply:
I've updated as per @ockham suggestion here https://github.com/WordPress/gutenberg/pull/57754/commits/1d3d2d63bc9b103682735fd56d44c1bcab5793a8 hopefully it is more clear now.